src/grimp_compare.rs

```diff
@@ -96,5 +99,58 @@ mod tests {
         let _ = build_imports_info("./data/large_graph.json")?;
 
         Ok(())
-    }   
+    }
+
+    #[test]
+    fn test_top_level_layers_large_graph() -> Result<()> {
+        let imports_info = build_imports_info("./data/large_graph.json")?;
+
+        let contract = LayeredArchitectureContract::new(&[
+            Layer::new([imports_info.package_info()._item("mypackage.data")], true),
+            Layer::new(
+                [imports_info.package_info()._item("mypackage.domain")],
+                true,
+            ),
+            Layer::new(
+                [imports_info.package_info()._item("mypackage.application")],
+                true,
+            ),
+            Layer::new(
+                [imports_info.package_info()._item("mypackage.plugins")],
+                true,
+            ),
+        ])
+        .with_deep_imports_allowed();
+
+        let result = contract.verify(&imports_info)?;
+        assert!(result.is_violated());
```

Comment:
This seems inconsistent with the results in grimp? https://github.com/seddonym/grimp/blob/81daa0dfdd0d2eb7aca611a42bc75f540d683db3/tests/benchmarking/test_benchmarking.py#L101